File: LeetCode/Math-probs/python/palindromeNumber.py
from dataclasses import dataclass
from enum import Enum
from typing import Tuple

class Result(Enum):
    PASS = "\033[92mPASS\033[00m"
    FAIL = "\033[91mFAIL\033[00m"

@dataclass
class palindromeNumberRecord:
    x: int
    expected: bool

class Solution:
    def isPalindrome1(self, x: int) -> bool:
        if x < 0:
            return False
        
        reverse: int = 0
        xcopy: int = x

        while x > 0:
            reverse = (reverse * 10) + (x % 10)
            x = x // 10

        return reverse == xcopy
    
    def isPalindrome2(self, x: int) -> bool:
        if x < 0:
            return False
        
        x: str = str(x)

        l = 0
        r = len(x) - 1

        while l < r:
            if x[l] != x[r]:
                return False
            l += 1
            r -= 1
        
        return True

    @staticmethod
    def testSolution(record: palindromeNumberRecord) -> None:
        print(f"input: {record.x}")
        print(f"expected: {record.expected}")

        res: bool = Solution().isPalindrome2(record.x)
        print(f"result: {res}")
        print(Result.PASS.value if res == record.expected else Result.FAIL.value)

if __name__ is "__main__":
    records: Tuple[palindromeNumberRecord] = (
        palindromeNumberRecord(121, True),
        palindromeNumberRecord(-121, False),
        palindromeNumberRecord(10, False)
    )

    for i, record in enumerate(records):
        print(f"# Test case {i + 1}")
        Solution.testSolution(record)
        print("-" * 50)


    

File: LeetCode/MathProbs/java/palindromeNumber/palindromeNumber.java
package LeetCode.MathProbs.java.palindromeNumber;

enum testResult {
    PASS("\u001B[32mPASS\u001B[0m"),
    FAIL("\u001B[31mFAIL\u001B[0m");

    private final String value;

    testResult(String value) {
        this.value = value;
    }

    public String getValue() {
        return this.value;
    }
}

record palindromeNumberRecord(int x, boolean expected) {
}

class palindromeNumber {
    public boolean isPalindrome1(int x) {
        if (x < 0) {
            return false;
        }

        String xString = String.valueOf(x);

        int l = 0;
        int r = xString.length() - 1;

        while (l < r) {
            if (xString.charAt(l) != xString.charAt(r)) {
                return false;
            }

            l++;
            r--;
        }

        return true;
    }

    public boolean isPalindrome2(int x) {
        if (x < 0) {
            return false;
        }

        int rev = 0;
        int xcopy = x;

        while (x > 0) {
            rev = (rev * 10) + (x % 10);
            x = x / 10;
        }
        return xcopy == rev;
    }

    public static void main(String[] args) {
        palindromeNumberRecord[] records = new palindromeNumberRecord[] {
                new palindromeNumberRecord(121, true),
                new palindromeNumberRecord(-121, false),
                new palindromeNumberRecord(10, false)
        };

        int i = 1;
        for (palindromeNumberRecord record : records) {
            System.out.println("# Test case " + i++);
            palindromeNumber.testSolution(record);
            System.out.println("-".repeat(50));
        }
    }

    private static void testSolution(palindromeNumberRecord record) {
        System.out.println("input: " + record.x());
        System.out.println("expected: " + record.expected());

        boolean res = new palindromeNumber().isPalindrome2(record.x());
        System.out.println("result: " + res);

        System.out.println(record.expected() == res ? testResult.PASS.getValue() : testResult.FAIL.getValue());
    }
}
